src/app/complex/complex-results/complex-results.component.ts

```diff
@@ -96,6 +110,10 @@ export class ComplexResultsComponent implements OnInit, AfterViewInit {
     return this._spicesFilter.length + this._interactorTypeFilter.length + this._bioRoleFilter.length;
   }
 
+  private doesComplexHaveComponent (complex, component): boolean{
```

Comment:
This method needs to be moved to `ComplexListComponent`, and we need to do `complex.componentIds.includes(component)`